packages/aws-rfdk/lib/deadline/scripts/bash/installDeadlineRepository.sh

```diff
@@ -95,12 +95,44 @@ fi
 SECRET_MANAGEMENT_ARG=''
 if [ "$SECRET_MANAGEMENT_ENABLED" == "true" ]; then
   sudo yum install -y jq
-  echo "Secret management is enabled. Credentials are stored in secret: $SECRET_MANAGEMENT_ARN"
+  
   SM_SECRET_VALUE=$(aws secretsmanager get-secret-value --secret-id=$SECRET_MANAGEMENT_ARN --region=$AWS_REGION)
   SM_SECRET_STRING=$(jq -r '.SecretString' <<< "$SM_SECRET_VALUE")
   SECRET_MANAGEMENT_USER=$(jq -r '.username' <<< "$SM_SECRET_STRING")
   SECRET_MANAGEMENT_PASSWORD=$(jq -r '.password' <<< "$SM_SECRET_STRING")
-  SECRET_MANAGEMENT_ARG="--installSecretsManagement true --secretsAdminName $SECRET_MANAGEMENT_USER --secretsAdminPassword $SECRET_MANAGEMENT_PASSWORD"
+  SECRET_MANAGEMENT_ARG="--installSecretsManagement true --secretsAdminName \"$SECRET_MANAGEMENT_USER\" --secretsAdminPassword \"$SECRET_MANAGEMENT_PASSWORD\""
+  len=$(echo ${#SECRET_MANAGEMENT_PASSWORD})
+  if test $len -ge 8 ; then
```

Comment:
We could change this to one if statement where all the conditions are compared with logical OR and then have a single error message stating all the password requirements when the password is not strong enough. I think this will make the code look a little cleaner, but it's not necessary though, just a suggestion.

Reply:
Thanks, this look better. I think we can make it even simpler though:

```bash
if [[ ${#SECRET_MANAGEMENT_PASSWORD} -lt 8 ]] || ! echo $1 | grep -e [0-9] | grep -e [a-z] | grep -e [A-Z] | grep -qe [^[:alnum:]]; then
    echo "ERROR: Admin password is too weak. It must be at least 8 characters long and contain at least one lowercase letter, one uppercase letter, one symbol and one digit."
    exit 1
fi
```

Reply:
I think this should work, it at least gets rid of the nested if statements:

```bash
if !([[ ${#SECRET_MANAGEMENT_PASSWORD} -ge 8 ]] && 
  echo $SECRET_MANAGEMENT_PASSWORD | grep -q [0-9] &&
  echo $SECRET_MANAGEMENT_PASSWORD | grep -q [a-z] &&
  echo $SECRET_MANAGEMENT_PASSWORD | grep -q [A-Z] &&
  echo $SECRET_MANAGEMENT_PASSWORD | grep -q [^[:alnum:]])
then
    echo "ERROR: Admin password is too weak. It must be at least 8 characters long and contain at least one lowercase letter, one uppercase letter, one symbol and one digit."
    exit 1
fi
```

Reply:
I like it. looks much better.